File: backend/app/schemas/alert.py
from datetime import datetime
from typing import Optional
from uuid import UUID

from pydantic import BaseModel

from app.schemas.template_measure import MeasureResponse


class AlertBase(BaseModel):
    patient_id: UUID
    treatment_session_id: UUID
    treatment_log_id: UUID
    measure_id: UUID
    template_measure_id: Optional[UUID] = None
    threshold_type: str
    threshold_value: float
    triggered_value: float
    message: str
    status: str


class AlertResponse(AlertBase):
    id: UUID
    created_at: datetime
    measure: Optional[MeasureResponse] = None

    class Config:
        from_attributes = True

File: frontend/src/types/alert.ts
import type { Measure } from "./template";

export interface Alert {
  id: string;
  patient_id: string;
  treatment_session_id: string;
  treatment_log_id: string;
  measure_id: string;
  template_measure_id?: string | null;
  threshold_type: string;
  threshold_value: number;
  triggered_value: number;
  message: string;
  status: string;
  created_at: string;
  measure?: Measure | null;
}
